packages/react-instantsearch-core/src/connectors/connectInfiniteHits.js

```diff
@@ -87,10 +113,16 @@ export default createConnector({
     });
   },
 
-  refine(props, searchState) {
+  refine(
+    props,
+    searchState,
+    event,
+    index = this._lastReceivedPage !== undefined
```

Comment:
We could have inline this inside the function body it would be simpler to understand.

Reply:
I inlined it, but I do not find it simpler to understand :innocent: